multiversx_sdk_core/proto/transaction_serializer.py

```diff
@@ -0,0 +1,93 @@
+from typing import Protocol
+
+import multiversx_sdk_core.proto.transaction_pb2 as ProtoTransaction
+from multiversx_sdk_core.address import AddressConverter
+from multiversx_sdk_core.constants import TRANSACTION_OPTIONS_TX_GUARDED
+
+
+class ITransaction(Protocol):
+    sender: str
+    receiver: str
+    gas_limit: int
+    chainID: str
+    gas_price: int
+    sender_username: str
+    receiver_username: str
+    nonce: int
+    amount: int
+    data: bytes
+    version: int
+    signature: bytes
+    options: int
+    guardian: str
+    guardian_signature: bytes
+
+
+class ProtoSerializer:
+    def serialize_transaction(self, transaction: ITransaction) -> bytes:
+        converter = AddressConverter()
+        receiver_pubkey = converter.bech32_to_pubkey(transaction.receiver)
+        sender_pubkey = converter.bech32_to_pubkey(transaction.sender)
+
+        proto_transaction = ProtoTransaction.Transaction()
+        proto_transaction.Nonce = transaction.nonce
+        proto_transaction.Value = self.serialize_transaction_value(transaction.amount)
+        proto_transaction.RcvAddr = bytes(receiver_pubkey)
+        proto_transaction.RcvUserName = transaction.receiver_username.encode()
+        proto_transaction.SndAddr = bytes(sender_pubkey)
+        proto_transaction.SndUserName = transaction.sender_username.encode()
+        proto_transaction.GasPrice = transaction.gas_price
+        proto_transaction.GasLimit = transaction.gas_limit
+        proto_transaction.Data = transaction.data
+        proto_transaction.ChainID = transaction.chainID.encode()
+        proto_transaction.Version = transaction.version
+        proto_transaction.Signature = transaction.signature
+        proto_transaction.Options = transaction.options
+
+        if self._is_guarded_transaction(transaction):
+            guardian_address = transaction.guardian
+            proto_transaction.GuardAddr = converter.bech32_to_pubkey(guardian_address)
+            proto_transaction.GuardSignature = transaction.guardian_signature
+
+        encoded_tx: bytes = proto_transaction.SerializeToString()
+
+        return encoded_tx
+
+    def _is_guarded_transaction(self, transaction: ITransaction) -> bool:
```

Comment:
Would be nice if this function and the one below would not be required (also for JS).

Reply:
Will be removed in the future.